internal/core/services/identity.go

```diff
@@ -735,30 +737,44 @@ func (i *identity) createIdentity(ctx context.Context, tx db.Querier, hostURL st
 		return nil, nil, fmt.Errorf("can't save identity: %w", err)
 	}
 
-	rhsPublishers, err := i.rhsFactory.BuildPublishers(ctx, reverse_hash.RHSMode(i.credentialStatusSettings.RHSMode), &kms.KeyID{
-		Type: kms.KeyTypeEthereum,
-		ID:   i.credentialStatusSettings.OnchainTreeStore.PublishingKeyPath,
-	})
+	rhsMode := reverse_hash.RHSMode(i.credentialStatusSettings.RHSMode)
+	rhsPublishers, err := i.rhsFactory.BuildPublishers(
+		ctx,
+		rhsMode,
+		&kms.KeyID{
+			Type: kms.KeyTypeEthereum,
+			ID:   i.credentialStatusSettings.OnchainTreeStore.PublishingKeyPath,
+		})
 	if err != nil {
 		return nil, nil, fmt.Errorf("can't create RHS publisher: %w", err)
 	}
 
 	if len(rhsPublishers) > 0 {
-		log.Info(ctx, "publishing state to RHS", "publishers", len(rhsPublishers))
-		for _, rhsPublisher := range rhsPublishers {
-			err := rhsPublisher.PublishNodesToRHS(ctx, []mtproof.Node{
-				{
-					Hash: identity.State.TreeState().State,
-					Children: []*merkletree.Hash{
-						claimsTree.Root(),
-						&merkletree.HashZero,
-						&merkletree.HashZero,
+		if rhsMode == reverse_hash.RHSModeOnChain && !i.autoPublishingToOnChainRHS {
+			log.Info(ctx, `
+			state info (hashes):
+				root: '%s',
+				ClaimsTreeRoot: '%s',
+				RevocationsTreeRoot: '%s',
+				RootOfRootsTreeRoot: '%s',
+		`, identity.State.TreeState().State, claimsTree.Root(), &merkletree.HashZero, &merkletree.HashZero)
```

Comment:
I guess that this log is not working. Probably it should be a log.Debug.  And the signature for issuer node log is 
log(ctx, "message", "key1", v1, "key2", v2,....) And usually logs are written in a single line because they are not relevant for the reader.

```suggestion
			log.Info(ctx, "state info (hashes):","root",  '%s', identity.State.TreeState().State, "ClaimsTreeRoot", claimsTree.Root()"... etc..
```